main.tf

```diff
@@ -483,6 +484,14 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = var.network_plugin_mode != "Overlay" || var.network_plugin == "azure"
       error_message = "When network_plugin_mode is set to Overlay, the network_plugin field can only be set to azure."
     }
+    precondition {
+      condition     = !(var.ebpf_data_plane == "cilium" && var.network_plugin != "azure")
```

Comment:
Could we simplify the condition to `var.ebpf_data_plan != "cilium" || var.network_plugin == "azure"`?